static/src/javascripts/projects/facia/modules/ui/live-blog-updates.js

```diff
@@ -0,0 +1,147 @@
+        var fakeUpdate = _.isUndefined(oldBlockDate);
+
+        fastdom.write(function () {
+            _.forEach(targets, function (element) {
+                var numNewBlocks = 0,
+
+                    blocksHtml = _.chain(blocks)
+                        .map(function (block, index) {
+                            if (numNewBlocks < numDisplayedBlocks
+                                && (block.publishedDateTime > oldBlockDate || (fakeUpdate && index === 0))) {
+                                block.isNew = true;
+                                numNewBlocks += 1;
+                            }
+                            return block;
+                        })
+                        .slice(0, numDisplayedBlocks + numNewBlocks)
+                        .map(function (block, index) {
+                            return renderBlock(articleId, block, index);
+                        })
+                        .value()
+                        .join(''),
+
+                    el = bonzo.create(template(blocksTemplate, {
+                        newBlocksHeight: numNewBlocks * blockHeightPx,
+                        blocksHtml: blocksHtml
+                    }));
+
+                bonzo(element).empty().addClass(blocksClassname).append(el);
+
+                if (numNewBlocks) {
+                    setTimeout(function () {
+                        bonzo(el).removeAttr('style');
+                    }, 1000);
+                }
+            });
+        });
+    }
+
+    function sanitizeBlocks(blocks) {
+        return _.filter(blocks, function (block) {
+            return block.id && block.publishedDateTime && block.body && block.body.length >= 10;
+        });
+    }
+
+    function pruneOldBlockDates(obj) {
+        return _.omit(obj, function (date) {
+            return !date || (new Date() - new Date(date)) / 3600000 > forgetAfterHours; // hours old
+        });
+    }
+
+    function updateBlocks() {
+        var oldBlockDates = storage.session.get(storageKey) || {};
+
+        _.forEach(elementsById, function (elements, articleId) {
+            ajax({
+                url: '/' + articleId + '.json?rendered=false',
+                type: 'json',
+                crossOrigin: true
+            })
+            .then(function (response) {
+                var blocks = response && sanitizeBlocks(response.blocks);
+
+                if (blocks && blocks.length) {
+                    renderBlocks(articleId, elements, blocks, oldBlockDates[articleId]);
+                    oldBlockDates[articleId] = blocks[0].publishedDateTime;
+                    storage.session.set(storageKey, pruneOldBlockDates(oldBlockDates));
+                }
+            });
+        });
+
+        if (refreshMaxTimes) {
+            refreshMaxTimes -= 1;
+            setTimeout(updateBlocks, refreshSecs * 1000);
+            refreshSecs = refreshSecs * refreshDecay;
+        }
+    }
+
+    function start() {
+        fastdom.read(function () {
```

Comment:
Don't think you need this `read` here actually. Just looking stuff up in the DOM doesn't have to be done in a `read` cycle. What does have to be done is anything that forces layout to be computed (such as looking at the height of an element).